Document user configuration

gtristan opened the pull request:
Completely document user configuration

gtristan commented:
This pull requests depends on first landing https://github.com/apache/buildstream/pull/1433

cs-shadow commented:
I understand this PR has other dependencies but just wanted to give it an overall +1. It'll be handy to not have to specify a project in such cases.

Commit:
_context.py: Remote `prompt` as a valid configuration key

This was originally introduced in 27ca65931d3cf6cf897b623f21ba2363f3c54029,
in order to configure whether the user wanted to be prompted for various
commands.

This has long been discarded and only the validity of the unused key remains.

Commit:
_frontend/widget.py: Format full element names in `bst show --format %{deps}`

We live in a junction aware world, let's not accidentally print out simple
element names for dependencies which might live across junction boundaries.

Commit:
doc/source/using_config.rst: Completely and thoroughly document user config.

This YAML was historically mostly only documented in the defaults, leaving
much explanation to be desired, now we document everything that can be
controlled by user configuration thoroughly.

Commit:
src/buildstream/data/userconfig.yaml: Remove some comments.

This used to serve as the main documentation for user config, but this
is now thoroughly documented in the docs, we only display this at the
end in order to show what the default configuration is.

gtristan commented:
Note: This depends on the previous remote service refactoring, since those also significantly modify the user configuration.

sstriker reviewed (commented):
Assuming #1438 ends up approved, +1.
